### DISCUSSION TITLE
Diagonal movement issue

### DISCUSSION THREAD

**Discussion by umitgoren** [ChebanovDD/MatchSweets#3] [Q&A] [answered]
Hi, thanks for  this a nice piece of work. However, I think I found a bug. 

The items don't always move (slide) as expected in slide down fill mode.

![m3](https://user-images.githubusercontent.com/15526496/168792620-64aca095-ff86-4c54-b476-e3149972da57.png)

## Minimal reproduction code and steps

1. Open the sample app in Unity.
2. Create a board with 2 diagonal empty tiles around the top-left of the board.
3. Create a match around the bottom-left corner of the board

You will that empty slots are not filled with the items that should slide diagonally.

**Comment by ChebanovDD** [comment, marked as answer]
Hi, thanks for your comment.

Actualy it was my desition to prevent this kind of slides. The `SlideDownFillStrategy` is provided just as an example, and you are free to write your own slide down logic.

If you want to enable all types of slides down, you can only check the side grid slots in the `CanDropDiagonally` method as shown below.

```csharp
private bool CanDropDiagonally(IGameBoard<IUnityGridSlot> gameBoard, IUnityGridSlot gridSlot,
    GridPosition direction, out GridPosition gridPosition)
{
    var sideGridSlot = gameBoard.GetSideGridSlot(gridSlot, direction);

    if (sideGridSlot is { NotAvailable: true })
    {
        return gameBoard.CanMoveDown(sideGridSlot, out gridPosition);
    }

    gridPosition = GridPosition.Zero;
    return false;
}
```

**Reply by umitgoren** [reply]
Thanks, I think this should be the default behaviour.

**Reply by ChebanovDD** [reply]
I'll consider making this behavior the default.
